2024/calcite-react/demo/src/App.tsx

```diff
@@ -1,31 +1,62 @@
+import { useEffect, useRef, useState } from "react";
+
 import "@esri/calcite-components/dist/components/calcite-shell";
-import "@esri/calcite-components/dist/components/calcite-shell-panel";
-import "@esri/calcite-components/dist/components/calcite-panel";
+import { CalciteShell } from '@esri/calcite-components-react';
+
+import WebMap from "@arcgis/core/WebMap";
+import MapView from "@arcgis/core/views/MapView";
 
-import { CalcitePanel, CalciteShell, CalciteShellPanel } from '@esri/calcite-components-react';
+import { Header } from "./Components/Header/Header";
+import { Panel } from "./Components/Panel/Panel";
 import { View } from "./Components/View/View";
 
-import config from "./config/application.json";
+interface AppProps {
+  webmap: string;
+  title: string;
+  panelHeading: string;
+}
 
-import esriConfig from "@arcgis/core/config";
-import { Panel } from "./Components/Panel/Panel";
+function App({ webmap, title, panelHeading }: AppProps) {
+  const viewRef = useRef(null) as React.RefObject<HTMLDivElement>;
```

Comment:
hmm, you are using `as ...` again - is there a reason?

```suggestion
  const viewRef = useRef<HTMLDivElement | null>(null);
```

Reply:
sorry, overlooked. Done.